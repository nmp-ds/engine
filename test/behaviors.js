import { setup } from './_helpers.js';
import { expect, test } from 'vitest';

setup();

test('appearance', async ({ uno }) => {
  const { css } = await uno.generate('appearance-none');
  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .appearance-none{appearance:none;-webkit-appearance:none;}"
  `);
});

test('will-change', async ({ uno }) => {
  const { css } = await uno.generate(['will-change-height', 'will-change-color']);
  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .will-change-color{will-change:color;}
    .will-change-height{will-change:height;}"
  `);
});

test('overscroll', async ({ uno }) => {
  const classes = ['overscroll-auto', 'overscroll-contain', 'overscroll-none', 'overscroll-y-auto', 'overscroll-y-contain', 'overscroll-y-none', 'overscroll-x-auto', 'overscroll-x-contain', 'overscroll-x-none'];
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('scroll', async ({ uno }) => {
  const { css } = await uno.generate(['scroll-auto', 'scroll-smooth']);
  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .scroll-auto{scroll-behavior:auto;}
    .scroll-smooth{scroll-behavior:smooth;}"
  `);
});
