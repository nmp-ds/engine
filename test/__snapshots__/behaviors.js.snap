// Vitest Snapshot v1

exports[`overscroll 1`] = `
"/* layer: default */
.overscroll-auto{overscroll-behavior:auto;}
.overscroll-contain{overscroll-behavior:contain;}
.overscroll-none{overscroll-behavior:none;}
.overscroll-x-auto{overscroll-behavior-x:auto;}
.overscroll-x-contain{overscroll-behavior-x:contain;}
.overscroll-x-none{overscroll-behavior-x:none;}
.overscroll-y-auto{overscroll-behavior-y:auto;}
.overscroll-y-contain{overscroll-behavior-y:contain;}
.overscroll-y-none{overscroll-behavior-y:none;}"
`;
